appengine-managed-runtime/src/main/java/com/google/apphosting/vmruntime/VmRuntimeFileLogHandler.java

```diff
@@ -102,4 +111,57 @@ private static void reloadLoggingProperties(LogManager logManager) {
       System.err.println(e.getClass().getName() + ": " + e.getMessage());
     }
   }
+
+  @Override
+  public void close() throws SecurityException {
+    super.close();
+    if (handlers.decrementAndGet() <= 0 ) {
+      queue.offer(null);
```

Comment:
Ah so much for testing!    Anyway, null is a poor poison pill!  I'll make a specific LogEvent instance that is the poison pill.